linera-core/src/client/mod.rs

```diff
@@ -3090,12 +3098,17 @@ where
 
         let mut certificates = Vec::new();
         loop {
+            let mutex = self.state().client_mutex();
+            let guard = mutex.lock_owned().await;
```

Comment:
Why not lock at the beginning of this method and release when we leave it?

Reply:
We could, but then we don't allow other things to happen in between at all. E.g. if there are many incoming messages to process, this could prevent users from doing other mutations in the node service in between.